ttnn/cpp/ttnn/tensor/layout/tensor_layout.cpp

```diff
@@ -122,33 +156,98 @@ size_t TensorLayout::compute_page_size_bytes(const Size& page_size) const {
     return page_config_.get_page_size_bytes(page_size, dtype_);
 }
 
+Size TensorLayout::compute_physical_shard_shape(const Size& logical_shard_shape) const {
+    // TODO: Alignment before sharding can describe padding along non-height or width dims
+    // However, this is not compatible with logcial sharding and we just leave it as nullopt
+    const int alignment_rank = static_cast<int>(alignment_.size());
+    TT_FATAL(alignment_rank <= 2, "Alignment {} must be rank 2 or less to compute physical shard shape", alignment_);
+
+    auto physical_shard_height = alignment_rank == 2 ? CMAKE_UNIQUE_NAMESPACE::round_up(logical_shard_shape.height(), alignment_[-2]) : logical_shard_shape.height();
+    auto physical_shard_width = alignment_rank >= 1 ? CMAKE_UNIQUE_NAMESPACE::round_up(logical_shard_shape.width(), alignment_[-1]) : logical_shard_shape.width();
```

Comment:
similarly to the above suggestion, you probably can do this
```suggestion
    auto physical_shard_height = CMAKE_UNIQUE_NAMESPACE::round_up(logical_shard_shape.height(), alignment_[-2]);
    auto physical_shard_width = CMAKE_UNIQUE_NAMESPACE::round_up(logical_shard_shape.width(), alignment_[-1]);
```